crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -73,14 +73,19 @@ impl DynamicScene {
         let type_registry = type_registry.read();
 
         for resource in &self.resources {
-            let registration = type_registry
-                .get_with_name(resource.type_name())
-                .ok_or_else(|| SceneSpawnError::UnregisteredType {
-                    type_name: resource.type_name().to_string(),
-                })?;
+            let type_info = resource.get_represented_type_info().ok_or_else(|| {
+                SceneSpawnError::NoRepresentedType {
+                    type_path: resource.reflect_type_path().to_string(),
+                }
+            })?;
+            let registration = type_registry.get(type_info.type_id()).ok_or_else(|| {
+                SceneSpawnError::UnregisteredButReflectedType {
+                    type_path: type_info.type_path().to_string(),
+                }
+            })?;
```

Comment:
Nit: Should these be let-else statements? I'm not sure which is more idiomatic haha